tests/testserver/server.py

```diff
@@ -39,6 +40,10 @@ def __init__(self, handler, host='localhost', port=0, requests_to_handle=1, wait
         self.ready_event = threading.Event()
         self.stop_event = threading.Event()
 
+        self.server_sock = self._create_socket_and_bind()
```

Comment:
`server_sock` was not previously an instance variable, and I wanted to be able to access it outside of `_handle_requests` (e.g. `__exit__`).